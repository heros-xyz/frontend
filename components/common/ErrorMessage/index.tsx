import { Box, Text, BoxProps } from "@chakra-ui/react";
import Link from "next/link";
import { If, Then } from "react-if";

interface ErrorMessageProps extends BoxProps {
  errorMessage: string | undefined;
  condition: boolean | string | undefined;
  link?: string;
  linkText?: string;
}
const ErrorMessage: React.FC<ErrorMessageProps> = ({
  errorMessage,
  condition,
  link,
  linkText,
  ...props
}) => {
  return (
    <If condition={condition}>
      <Then>
        <Box textAlign="left" {...props}>
          <Text
            as="span"
            fontSize={{ base: "xs", lg: "md" }}
            color="error.dark"
          >
            {errorMessage}
          </Text>
          <If condition={!!link}>
            <Then>
              <Text
                fontSize={{ base: "xs", lg: "md" }}
                color="acccent.3"
                textDecoration="underline"
                mb={{ base: "1px", lg: 0 }}
                ml="4px"
                cursor="pointer"
              >
                <Link href={link ?? ""}>{linkText}</Link>
              </Text>
            </Then>
          </If>
        </Box>
      </Then>
    </If>
  );
};

export default ErrorMessage;
