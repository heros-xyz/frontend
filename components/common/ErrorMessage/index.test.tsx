import { render } from "@testing-library/react";
import ErrorMessage from ".";

it("renders Error Message unchanged", () => {
  const { container } = render(
    <ErrorMessage condition={true} errorMessage={"Field is required"} />
  );
  expect(container).toMatchSnapshot();
});
