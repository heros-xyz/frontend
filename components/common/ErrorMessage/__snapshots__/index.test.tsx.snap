// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Error Message unchanged 1`] = `
<div>
  <div
    class="css-13brihr"
  >
    <span
      class="chakra-text css-1kn7aai"
    >
      Field is required
    </span>
  </div>
</div>
`;
